p2p/utils.py

```diff
@@ -33,6 +36,25 @@ def get_devp2p_cmd_id(msg: bytes) -> int:
     return rlp.decode(msg[:1], sedes=rlp.sedes.big_endian_int)
 
 
+@to_tuple
+def sequence_builder(start_number: int, max_length: int, skip: int, reverse: bool) -> Iterator[int]:
+    remaining_elements = max_length
+    next_element = start_number
+    step_size = skip + 1
+
+    while remaining_elements > 0:
```

Comment:
We could drop the loop mutation and just do `for _ in range(max_length)`.  That should drop the inner `iff/else` as well as the need to decrement `remaining_elements` in each loop.

Reply:
:+1: for dropping the mutation, that was embarrassing : P

We still need the if check for dropping below 0, for example:

```py
@pytest.mark.parametrize(
    'start_num, max_length, skip, reverse, expected',
    (
        ...,
        (1, 9, 0, True, (1, 0)),
    ),
)       
def test_sequence(start_num, max_length, skip, reverse, expected):
    assert sequence_builder(start_num, max_length, skip, reverse) == expected
```

Are you talking about a different `iff/else`?

Reply:
> We still need the if check for dropping below 0, for example:

I don't know what I was talking about... carry on and :shipit: